packages/tests/server/minitrpc.ts

```diff
@@ -0,0 +1,98 @@
+import type {
+  AnyMutationProcedure,
+  AnyProcedure,
+  AnyQueryProcedure,
+  AnyRouter,
+  ProcedureArgs,
+  ProcedureRouterRecord,
+} from '@trpc/server';
+import type { TRPCResponse } from '@trpc/server/rpc';
+import type { inferTransformedProcedureOutput } from '@trpc/server/shared';
+
+interface ProxyCallbackOptions {
+  path: string[];
+  args: unknown[];
+}
+
+type ProxyCallback = (opts: ProxyCallbackOptions) => unknown;
+
+const noop = () => {
+  // noop
+};
+
+function createRecursiveProxy(callback: ProxyCallback, path: string[]) {
```

Comment:
We can link that we have a version if this in the real codebase

https://github.com/trpc/trpc/blob/main/packages/server/src/shared/createProxy/index.ts